### PR TITLE
Fix nested object in array #3

### LINKED ISSUE #3: A nested object in array is not converted
Hi, abouthiroppy.
Thank you for your 'convert-keys'. This is a great library.

I’ve read your ‘convert-keys’ codes and noticed that a nested object test will fail.
Is this a spec or bug?
This is an example of re-produce for this issue.

``` js
test('should return object as coverted camel case', async (t) => {
  const obj = {
    'hoge_fuga': 1,
    'hogeHoge': {
      'piyo_piyo': 'aaaa',
      'fuga_fuga': 'bbbb'
    },
    'piyo_array': [{ 'foo_bar': 'a'}, {'foo_bar': 'b' }]
  };

  const res = {
    hogeFuga: 1,
    hogeHoge: {
      piyoPiyo: 'aaaa',
      fugaFuga: 'bbbb'
    },
    piyoArray: [{ fooBar: 'a'}, {fooBar: 'b' }] // will fail here
  };

  t.deepEqual(convertKeys.toCamel(obj), res);
});
```

### PR DESCRIPTION
**Pull request by jyane**
I have fixed #3, and added corner case tests.
NOTE: `walk` is now covariant function.

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -31,7 +47,11 @@ test('should return object as coverted snake case', async (t) => {
       'piyo_piyo': 'aaaa',
       'fuga_fuga': 'bbbb',
       'testTest': 'test'
-    }
+    },
+    'piyo_array': [
+      {'fooBar': 'a'},
+      {'fooBar': 'b'}
```
Comment (hiroppy) [inline test/index.js:53]: Could you please delete `'` of key?

**Review by hiroppy — CHANGES_REQUESTED** [review]

File: test/index.js (hunk 1 above)
Comment (jyane) [reply test/index.js:53]: Is it only #51-#53 ?

Comment (hiroppy) [reply test/index.js:53]: Ah, I see...(Seems to be at many places.) OK, I will fix next time 👍

**Review by hiroppy — APPROVED** [review]

**Comment by hiroppy** [comment]
Awesome! Thanks a lot:)
